rero_ils/modules/documents/views.py

```diff
@@ -717,29 +720,19 @@ def online_holdings(document_pid, viewcode='global'):
 
 
 @blueprint.app_template_filter()
-def subject_format(subject, language):
+def subject_format(subject_data, language):
     """Format the subject according to the available keys.
 
-    :param subject: the record subject.
+    :param subject_data: the record subject.
     :param language: current language on interface.
     """
-    for key in ['$ref', 'term', 'preferred_name', 'title']:
-        value = subject.get(key)
-        # key does not exists try the next one
-        if not value:
-            continue
-        # resolve $ref and retrieve the name in the given language
-        if key == '$ref':
-            sub, _ = Contribution.get_record_by_ref(value)
-            return sub._get_mef_localized_value(
-                'preferred_name', language)
-        # add the creator to the title
-        creator = subject.get('creator')
-        if key == 'title' and creator:
-            value = ' / '.join([value, creator])
-        # do nothing for term and preferred_name
-        if value:
-            return value
+    try:
+        return SubjectFactory \
+            .create_subject(subject_data) \
+            .render(language=language)
+    except (AttributeError, TypeError) as err:
+        # print(str(err))
```

Comment:
could be removed ?

Reply:
It could. But, as other places into the code, it's a good checkpoint for debugging code if needed. 
Just uncomment this line and run unitest will tells you a lot about your error.